### PR TITLE
Added Spring-06 HW implementation

### PR DESCRIPTION
**Pull request by oroboro-cpu**

### CODE REVIEW

File: src/main/java/cinema/project/controller/AuthenticationController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
+public class AuthenticationController {
+    private final AuthenticationService service;
+    private final UserMapper userMapper;
```
Comment (Alekearl) [inline src/main/java/cinema/project/controller/AuthenticationController.java]: Do you realy need this field here, where you using it here?

File: src/main/java/cinema/project/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @GetMapping
+    public List<OrderResponseDto> getHistory(@RequestParam Long userId) {
+        return orderService.getOrdersHistory(userService.get(userId)).stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
```
Comment (Alekearl) [inline src/main/java/cinema/project/controller/OrderController.java]: Here better to use void method, cause we just posting and don't need smth to return.

File: src/main/java/cinema/project/dao/UserDaoImpl.java [hunk 3]
```diff
@@ -51,4 +51,13 @@ public User add(User user) {
                     + email, ex);
         }
     }
+
+    @Override
+    public Optional<User> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(User.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find user bi id: " + id, e);
```
Comment (Alekearl) [inline src/main/java/cinema/project/dao/UserDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't find user by id: " + id, e);
```

File: src/main/java/cinema/project/model/dto/response/ShoppingCartResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package cinema.project.model.dto.response;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
+    private Long userId;
```
Comment (Alekearl) [inline src/main/java/cinema/project/model/dto/response/ShoppingCartResponseDto.java:8]: I think there should be enough userShoppingCartId

File: src/main/java/cinema/project/service/mapper/UserMapper.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    @Override
+    public UserResponseDto toDto(User user) {
+        UserResponseDto userResponseDto = new UserResponseDto();
+        userResponseDto.setId(user.getId());
+        userResponseDto.setEmail(user.getEmail());
+        return userResponseDto;
+    }
+
+    @Override
+    public User toEntity(UserRequestDto userRequestDto) {
```
Comment (Alekearl) [inline src/main/java/cinema/project/service/mapper/UserMapper.java]: Do you use this method somewhere?

**Review by Alekearl — COMMENTED** [review]
Few suggestions

**Review by Alekearl — APPROVED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**